app/uk/gov/hmrc/gform/formtemplate/FormTemplateValidator.scala

```diff
@@ -149,14 +149,14 @@ object FormTemplateValidator {
     }
   }
 
-  val userContextComponentType: List[FormComponent] => List[FormComponent] =
+  val userContextComponentType: List[FormComponent] => Boolean =
     enrolledIdentifierComponents =>
       enrolledIdentifierComponents.collect {
```

Comment:
More clear way would be to use `enrolledIdentifierComponents.exists(...)`. There is no need to filter/map whole list and asking if result list is empty.